tests/ui_tests.cpp

```diff
@@ -105,10 +117,19 @@ void check_testcase(const testcase_t &tc) {
     }
     std::cout << std::endl << std::endl;
 
-    EXPECT_EQ(output.size(), tc.expected.size());
-    for (size_t i = 0; i < tc.expected.size(); i++) {
-        if (i < output.size()) {
-            EXPECT_THAT(output[i], testing::Eq(tc.expected[i]));
+    if (app_mode_expert()) {
+        EXPECT_EQ(output.size(), tc.expected_expert.size());
+        for (size_t i = 0; i < tc.expected_expert.size(); i++) {
+            if (i < output.size()) {
+                EXPECT_THAT(output[i], testing::Eq(tc.expected_expert[i]));
+            }
+        }
+    } else {
+        EXPECT_EQ(output.size(), tc.expected.size());
+        for (size_t i = 0; i < tc.expected.size(); i++) {
+            if (i < output.size()) {
+                EXPECT_THAT(output[i], testing::Eq(tc.expected[i]));
+            }
         }
     }
 }
```

Comment:
We can probably avoid code duplication by doing something like

```
var expected = app_mode_expert() ? tc.expected_expert : tc.expected;
```

and then all other code remains a single copy

Reply:
I like it! I will do it.